.github/workflows/build.yml

```diff
@@ -66,7 +66,32 @@ jobs:
         working-directory: ${{ github.workspace }}/packages
       - uses: actions/upload-artifact@v2
         with:
-          name: libddwaf-macos
+          name: libddwaf-macos-x86_64
+          path: ${{ github.workspace }}/packages
+
+  macos-cross-build:
```

Comment:
This is mostly a dupe of `macos-build`, except:

- it adds a `cmake` variable to cross compile
- it skips test building: one less target on the build step; it does build but attempts to run the binary right after which makes it fail
- it skips the test running step altogether

Maybe that can be folded into `macos-build` with a matrix and a couple of variables + conditional steps.